storage/innobase/include/trx0trx.h

```diff
@@ -902,6 +903,146 @@ struct TrxVersion {
 
 typedef std::list<TrxVersion, ut_allocator<TrxVersion> > hit_list_t;
 
+/** Utility class to collect and report slow query log InnoDB extension
+stats */
+class trx_stats {
+	public:
+	trx_stats() {
+	/* Always created in a zeroed memory block */
+#ifdef UNIV_DEBUG
+		ut_ad(lock_que_wait_ustarted == 0);
+		ut_ad(take_stats == false);
+#endif
+	}
+
+	/**
+	Register, if needed, start of an I/O read or wait.
+
+	@param	trx	transaction to account I/O to or NULL
+	@param	bytes	number of bytes to read or 0 if this is a wait for an
+	 already posted read in progress
+	@return value to be passed to end_io_read
+	*/
+	MY_NODISCARD
+	static ib_uint64_t
+	start_io_read(trx_t *trx, ulint bytes);
```

Comment:
Also consider an auxiliary RAII class which would do `start_io_read()` in constructor and `end_io_read()` in destructor.